File: Ejercicios Propuestos 3/Ejercicio 6/Ejercicio 6.py
NAME = "Renzo Leonardo Cienfuegos Cardenas"
GROUP = 2

# Solicitar la cantidad de caracteres
n = int(input("Ingrese la cantidad de caracteres: "))

# Crear una lista vacía para almacenar los caracteres
cadena = []

# Leer los caracteres uno por uno
for i in range(n):
    caracter = input(f"Carácter para la posición {i}: ")
    cadena.append(caracter)

# Bubble Sort
for i in range(n - 1):
    for j in range(n - i - 1):
        # Comparar los caracteres consecutivos
        if cadena[j] > cadena[j + 1]:
            # Intercambiar los caracteres
            aux = cadena[j]
            cadena[j] = cadena[j + 1]
            cadena[j + 1] = aux

# Mostrar la cadena ordenada
print("Cadena ordenada:")

for caracter in cadena:
    print(caracter, end="")

print()  # Salto de línea

File: Ejercicios Propuestos 3/Ejercicio 6/Ejercicio 6.jl
print("Ingrese la cantidad de caracteres: ")
n = parse(Int, readline())

cadena = Vector{Char}(undef, n)

for i in 0:(n - 1)
    print("Carácter para la posición ", i, ": ")
    
    entrada = readline()
    cadena[i + 1] = isempty(entrada) ? ' ' : entrada[1]
end

# Bubble Sort
for i in 0:(n - 2)
    for j in 0:(n - i - 2)
        #  sumamos 1 a los índices j y j+1 para adaptarlos a Julia
        if cadena[j + 1] > cadena[(j + 1) + 1]
            aux = cadena[j + 1]
            cadena[j + 1] = cadena[(j + 1) + 1]
            cadena[(j + 1) + 1] = aux
        end
    end
end

println("Cadena ordenada:")
for i in 0:(n - 1)
    # print() escribe "Sin Saltar"
    print(cadena[i + 1])
end
println("") # Salto de línea final
